### PR TITLE
HOSTEDCP-2063: [CPO Refactor] Refactor openshift-apiserver

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/v2/oapi/deployment.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+				})
+			}
+		}
+	})
+
+	if cpContext.HCP.Spec.Configuration.GetAuditPolicyConfig().Profile == configv1.NoneAuditProfileType {
+		util.RemoveContainer("audit-logs", &deployment.Spec.Template.Spec)
+	}
+
+	serviceServingCA, err := getServiceServingCA(cpContext)
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/v2/oapi/deployment.go:84]: have we thought about a more explicit way to signal this dependencies?

**Comment by muraee** [comment]
/override "Red Hat Konflux / hypershift-operator-main-on-pull-request"

**Comment by muraee** [comment]
/override "Red Hat Konflux / hypershift-operator-main-on-pull-request"

**Pull request merged**